lib/ami-existence-poller.js

```diff
@@ -0,0 +1,56 @@
+const assert = require('assert');
+const {runAWSRequest} = require('./aws-request');
+const Iterate = require('taskcluster-lib-iterate');
+
+const log = require('./log');
+
+const BATCH_SIZE = 100;
+
+class AmiExistencePoller {
+
+  constructor({ec2, runaws = runAWSRequest, amiPollDelay = 25, state}) {
+    assert(typeof ec2 === 'object');
+    assert(typeof state === 'object');
+
+    this.ec2 = ec2;
+    this.state = state;
+    this.runaws = runaws;
+
+    this.iterator = new Iterate({
+      maxIterationTime: 1000 * 60 * 15,
+      watchDog: 1000 * 60 * 15,
+      maxFailures: 10,
+      waitTime: amiPollDelay,
+      handler: async (watchdog) => {
+        try {
+          await this.poll();
```

Comment:
It looks like this is missing an argument to the function.  That seems right, since we want to do batches of ami ids at the same time, but as implemented, this is going to throw an assertion failure on the first call.